### PR TITLE
Adding the ability to filter reviews and requests by name

### PR DESCRIPTION
**Pull request by neil-power-hs**
Allow filtering of reviews and review requests through query parameters.

### CODE REVIEW

File: backend/src/main/kotlin/com/hootsuite/hermes/Main.kt [hunk 1]
```diff
@@ -210,10 +210,16 @@ suspend fun registerTeamGet(call: ApplicationCall) {
...
-    val reviewRequests = transaction { ReviewRequestEntity.all().joinToString("<br>") { it.toString() } }
+    val requestsString = transaction {
+        val name = call.request.queryParameters["name"]
+        val requests = ReviewRequestEntity.all()
+        val filteredRequests = name?.let { requests.filter { it.githubName == name } } ?: requests
+        // TODO Do this on the frontend when built
+        filteredRequests.joinToString("<br>") { "<a href=\"${it.htmlUrl}\">$it</a>" }
+    }
     val teamsHtml = StringBuilder()
     teamsHtml.append("<h1>Review Requests</h1><p>")
```
Comment (benjamin-hergert-hs) [inline backend/src/main/kotlin/com/hootsuite/hermes/Main.kt]: I'd prefer templates for this kind of stuff with a multiline string. `StringBuilder` seems unnecessary here.
I know that they are just for testing, but if they are to stick around in the long run, I'd treat them as production code.

File: backend/src/main/kotlin/com/hootsuite/hermes/Main.kt [hunk 2]
```diff
@@ -224,10 +230,16 @@ suspend fun reviewRequestsGet(call: ApplicationCall) {
...
 suspend fun reviewsGet(call: ApplicationCall) {
-    val reviews = transaction { ReviewEntity.all().joinToString("<br>") { it.toString() } }
+    val reviewsString = transaction {
+        val name = call.request.queryParameters["name"]
+        val reviews = ReviewEntity.all()
+        val filteredReviews = name?.let { reviews.filter { it.githubName == name }} ?: reviews
+        // TODO Do this on the frontend when built
+        filteredReviews.joinToString("<br>") { "<a href=\"${it.htmlUrl}\">$it</a>" }
+    }
     call.respondText(StringBuilder().apply {
```
Comment (benjamin-hergert-hs) [inline backend/src/main/kotlin/com/hootsuite/hermes/Main.kt]: Same here

**Comment by neil-power-hs** [comment]
!hermes rejected

**Comment by neil-power-hs** [comment]
!hermes

**Comment by neil-power-hs** [comment]
!hermes @benjamin-hergert-hs

**Pull request merged**